### ISSUE TITLE
Lesson for making a TupleTool

### ISSUE THREAD

**Issue by alexpearce** [lhcb/first-analysis-steps#89]
We should have a lesson for making a basic TupleTool. Provide a nicely formatted, sane template that adds some simple variables.

Could leave using the tool in a `DecayTreeTuple` instance as a challenge.

**Comment by apuignav** [comment]
Idea: isolation variable that counts how many vertices you can make by adding one extra track to the vertex which have a difference in chi2 smaller than a configurable number.

**Comment by saschastahl** [comment]
I have some old code which does something very similar. It adds as many tracks as possible to a given vertex until a certain chi2 is reached.

**Comment by betatim** [comment]
What is the aim of this? Teaching people to do some first coding for LHCb or TupleTooling? I would suggest we need something for first coding first. TupleTool could do that job but then we should choose something that is fairly easy to do, so that people only have problems with the mechanics of getting the code to compile. Then as a challenge maybe make the tool more realistic/better/trickier.

**Comment by saschastahl** [comment]
The ideas was to get used to code in the LHCb environment. A tuple tool would be a nice start as this is related to the other topics of the workshop.

**Comment by betatim** [comment]
Good idea. However also see my comment in #83: https://github.com/lhcb/first-analysis-steps/pull/83#issuecomment-108262888

I think we should remind ourselves of what is beginner stuff and what is medium stuff and start a new repo for those things instead of continually adding more here. I worry about mission creep.

**Comment by saschastahl** [comment]
Yes, adding this lesson was based on feedback by people. But we should probably discuss that in another thread.
I can still try to put some stuff together or is the opinion to not do that?
